hive/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -0,0 +1,120 @@
+  }
+
+  @Override
+  public org.apache.iceberg.Table createTable(
+      TableIdentifier identifier, Schema schema, PartitionSpec spec, Map<String, String> properties) {
+    Preconditions.checkArgument(identifier.namespace().levels().length == 1,
+        "Missing database in table identifier: %s", identifier);
+    return super.createTable(identifier, schema, spec, properties);
+  }
+
+  @Override
+  public org.apache.iceberg.Table loadTable(TableIdentifier identifier) {
+    Preconditions.checkArgument(identifier.namespace().levels().length == 1,
+        "Missing database in table identifier: %s", identifier);
+    return super.loadTable(identifier);
+  }
+
+  @Override
+  public boolean dropTable(TableIdentifier identifier) {
+    Preconditions.checkArgument(identifier.namespace().levels().length == 1,
+        "Missing database in table identifier: %s", identifier);
+    String database = identifier.namespace().level(0);
+
+    try {
+      clients.run(client -> {
+        client.dropTable(database, identifier.name());
+        return null;
+      });
+
+      return true;
+
+    } catch (NoSuchObjectException e) {
+      return false;
+
+    } catch (TException e) {
+      throw new RuntimeException("Failed to drop " + identifier.toString(), e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to dropTable", e);
+    }
+  }
+
+  @Override
+  public void renameTable(TableIdentifier from, TableIdentifier to) {
+    Preconditions.checkArgument(from.namespace().levels().length == 1,
+        "Missing database in table identifier: %s", from);
+    Preconditions.checkArgument(to.namespace().levels().length == 1,
+        "Missing database in table identifier: %s", to);
+
+    String toDatabase = to.namespace().level(0);
+    String fromDatabase = from.namespace().level(0);
+    String fromName = from.name();
+
+    try {
+      Table table = clients.run(client -> client.getTable(fromDatabase, fromName));
+      table.setDbName(toDatabase);
+      table.setTableName(to.name());
+
+      clients.run(client -> {
+        client.alter_table(fromDatabase, fromName, table);
+        return null;
+      });
+
+    } catch (TException e) {
+      throw new RuntimeException("Failed to rename " + from.toString() + " to " + to.toString(), e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to rename", e);
+    }
+  }
+
+  @Override
+  public TableOperations newTableOps(
+      Configuration configuration, TableIdentifier tableIdentifier) {
+    String dbName = tableIdentifier.namespace().level(0);
+    String tableName = tableIdentifier.name();
+    return new HiveTableOperations(configuration, clients, dbName, tableName);
+  }
+
+  protected String defaultWarehouseLocation(Configuration hadoopConf, TableIdentifier tableIdentifier) {
```

Comment:
Would it also make sense to expose a property that allows users to set a custom table location?
Today, Spark uses `LOCATION` to control if the data source table is managed/external.
Will we be able to use `CREATE TABLE t ... USING iceberg LOCATION customLocation` with Hive tables?

Reply:
Yes, in Spark we're passing this as "location" and should probably support it here.

Reply:
I've added support for passing location into `createTable`. The Spark integration should pass the `location` table property there.